src/app/operators/dialogs/box-plot-operator/box-plot-operator.component.ts

```diff
@@ -104,14 +137,11 @@ export class BoxPlotComponent implements OnInit, AfterViewInit {
     }
 
     updateAttribute(i: number, attribute: string) {
-        // console.log(i, attribute);
-        const attributes = this.form.controls['attributes'].value;
-        const newAttributes = [];
-        for (let j = 0; j < attributes.length; j++) {
-            newAttributes[j] = attributes[j];
-        }
-        newAttributes[i] = attribute;
-        this.form.controls['attributes'].setValue(newAttributes);
+        this.selectedLayers.first().subscribe(selectedLayers => {
```

Comment:
You use the same method in `onInit` and here and always with first. A suggestion: Do not use `first` but keep the subscription (remember to unsubscribe `onDestroy`).